File: src/test/java/FooBarQixTest.java
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class FooBarQixTest {

    @Test
    public void should_write_foo_if_number_can_be_divided_by_3(){
        //ARRANGE
        String expected = "Foo";
        //ACT
        String result = FooBarQix.compute(3);
        //ASSERT
        assertThat(result).isEqualTo(expected);
    }

    @Test
    public void should_write_bar_if_number_can_be_divided_by_5(){
        //ARRANGE
        String expected = "Bar";
        //ACT
        String result = FooBarQix.compute(10);
        //ASSERT
        assertThat(result).isEqualTo(expected);
    }

    @Test
    public void should_write_bar_if_number_can_be_divided_by_7(){
        //ARRANGE
        String expected = "Qix";
        //ACT
        String result = FooBarQix.compute(7);
        //ASSERT
        assertThat(result).isEqualTo(expected);
    }

    @Test
    public void should_write_FooQix_if_number_can_be_divided_by_3_and_7(){
        //ARRANGE
        String expected = "FooQix";
        //ACT
        String result = FooBarQix.compute(21);
        //ASSERT

        assertThat(result).isEqualTo(expected);
    }

    @Test
    public void should_write_FooBarBar_if_number_can_be_divided_by_3_and_5_and_contains_5(){
        //ARRANGE
        String expected = "FooBarBar";
        //ACT
        String result = FooBarQix.compute(15);
        //ASSERT

        assertThat(result).isEqualTo(expected);
    }


}

File: src/test/java/FooBarQixTest.kt
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

import java.util.Arrays

import org.assertj.core.api.Assertions.assertThat

@RunWith(Parameterized::class)
class FooBarQixTest(private val expectedResult: String, private val inputNumber: Int) {


    @Test
    fun isComputingCorrectlyTest() {
        val result = FooBarQix.compute(inputNumber)
        assertThat(result).isEqualTo(expectedResult)
    }

    companion object {

        @JvmStatic
        @Parameterized.Parameters
        fun params(): Collection<Array<Any>> {
            return Arrays.asList(
                    arrayOf("FooFoo", 3),
                    arrayOf("Bar", 10),
                    arrayOf("QixQix", 7),
                    arrayOf("BarBar", 5),
                    arrayOf("FooQix", 21),
                    arrayOf("FooBarBar", 15),
                    arrayOf("FooBarQixBar", 105),
                    arrayOf("Qix", 14),
                    arrayOf("Foo", 9),
                    arrayOf("FooFooFoo", 33),
                    arrayOf("BarBarBar", 55),
                    arrayOf("QixQixQix", 77)

            )
        }
    }


}